### PR TITLE
2.5.1

### LINKED ISSUE #152: infinit loop for session delete request
under to high load from time to time we see the situation when grid continue to send delete requests for non-existing session:
```
2024-08-12 13:20:43:315 - [HTTP] <-- HEAD /wd/hub/status 200 2 ms - 110
2024-08-12 13:20:43:315 - [HTTP]
2024-08-12 13:20:46:479 - [HTTP] --> DELETE /wd/hub/session/3af99f3b-8967-46e2-b5a5-1afe53e407d1
2024-08-12 13:20:46:480 - [HTTP] {}
2024-08-12 13:20:46:484 - [HTTP] <-- DELETE /wd/hub/session/3af99f3b-8967-46e2-b5a5-1afe53e407d1 404 4 ms - 2648
```

as result device is not used during the regression and new requests hanags and fail

### LINKED ISSUE #153: Block wda/adb requests for device for 1 minute between tries

### PR DESCRIPTION
**Pull request by akamarouski**

### CODE REVIEW

File: src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java [hunk 1]
```diff
@@ -78,6 +82,24 @@ protected Boolean initialize() throws ConcurrentException {
...
         }
     };
+    public static final Map<String, Duration> DEVICE_IGNORE_AUTOMATION_TIMERS = new ConcurrentHashMap<>();
+
+    private static final Duration DOWN_TIMEOUT = Optional.ofNullable(System.getenv("DOWN_TIMEOUT"))
+            .filter(StringUtils::isNotBlank)
+            .map(Integer::parseInt)
+            .map(Duration::ofSeconds)
+            .orElse(Duration.ofMinutes(3));
+    private static final Duration UNHEALTHY_WDA_ADB_TIMEOUT = Optional.ofNullable(System.getenv("UNHEALTHY_WDA_ADB_TIMEOUT"))
```
Comment (vdelendik) [inline src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java:92]: let's rename to UNHEALTHY_MOBILE_TIMEOUT and just a comment regarding adb and wda

File: src/main/java/com/zebrunner/mcloud/grid/integration/client/STFClient.java [hunk 2]
```diff
@@ -60,29 +61,22 @@ public final class STFClient {
             .map(Duration::ofSeconds)
             .orElse(Duration.ofMinutes(5));
 
+    private static final Duration STF_DEVICE_MANUALLY_RESERVED_TIMEOUT = Optional.ofNullable(System.getenv("STF_DEVICE_MANUALLY_RESERVED_TIMEOUT"))
```
Comment (vdelendik) [inline src/main/java/com/zebrunner/mcloud/grid/integration/client/STFClient.java:64]: what sense of this timeout?

File: src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java [hunk 3]
```diff
@@ -78,6 +82,24 @@ protected Boolean initialize() throws ConcurrentException {
...
+            .filter(StringUtils::isNotBlank)
+            .map(Integer::parseInt)
+            .map(Duration::ofSeconds)
+            .orElse(Duration.ofMinutes(3));
+
+    private static final Duration INACTIVITY_RELEASE_TIMEOUT = Optional.ofNullable(System.getenv("INACTIVITY_RELEASE_TIMEOUT"))
+            .filter(StringUtils::isNotBlank)
+            .map(Integer::parseInt)
+            .map(Duration::ofSeconds)
+            .orElse(Duration.ofMinutes(10));
```
Comment (vdelendik) [inline src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java:102]: why 10 min?

File: src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java (hunk 1 above)
Comment (akamarouski) [reply src/main/java/com/zebrunner/mcloud/grid/MobileRemoteProxy.java]: done

**Pull request merged**